pkg/reader/helpers.go

```diff
@@ -0,0 +1,118 @@
+package reader
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	typeconv "github.com/smartcontractkit/chainlink-ccip/internal/libs/typeconv"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/contractreader"
+)
+
+// bindable is a helper interface to represent all different types of contract readers.
+// We expose per-reader type functions, but then cast to this common interface to avoid code duplication.
+type bindable interface {
```

Comment:
I couldn't find a better way to do it. It's because we have contractreaderFacade and contracreaderExtended which are different interfaces. I created dedicated function which only convert that to common type and execute the shared logic @winder

Reply:
This is a pretty confusing set of helpers

Reply:
This is nice, thanks.